### PR TITLE
fixed the sidebar

### PR DESCRIPTION
**Pull request by Roksolyanka**
Fixed the opening and closing of the sidebar.

### CODE REVIEW

File: src/components/Sidebar/Burger/index.tsx [hunk 1]
```diff
@@ -1,23 +1,27 @@
-import { useContext } from 'react';
-import { StyledBurger } from './styles';
-import { SidebarContext } from '../../../store/sidebar-context';
+import { useContext, useEffect, useState } from "react";
+import { StyledBurger } from "./styles";
+import { SidebarContext } from "../../../store/sidebar-context";
```
Comment (StareMisto) [inline src/components/Sidebar/Burger/index.tsx]: please, use single quotes

File: src/components/Sidebar/Burger/index.tsx [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
-  if (open) {
-    document.body.style.position = 'fixed';
-    document.body.style.top = `-${window.scrollY}px`;
-  } else {
-    document.body.style.position = '';
-    document.body.style.top = '';
-  }
+  useEffect(() => {
+    if (open) {
+      // Зберігаємо поточну позицію прокрутки при відкритті сайдбару
```
Comment (StareMisto) [inline src/components/Sidebar/Burger/index.tsx]: please, remove all comments, we leave comments only with prefix TODO:
for examle
//TODO: remove after testing

File: src/components/Sidebar/styles.ts [hunk 3]
```diff
@@ -13,18 +13,16 @@ export const SidebarStyled = styled.nav<SidebarStyledProps>`
     height: 100vh;
     justify-content: center;
     padding: 2rem;
-    position: absolute;
-    right: 0;
+    position: fixed;
+    right: ${({ open }) => (open ? "0" : "-100%")};
```
Comment (StareMisto) [inline src/components/Sidebar/styles.ts]: singe quotes

File: src/components/Sidebar/Burger/index.tsx [hunk 4]
```diff
@@ -1,23 +1,27 @@
-import { useContext } from 'react';
-import { StyledBurger } from './styles';
-import { SidebarContext } from '../../../store/sidebar-context';
+import { useContext, useEffect, useState } from "react";
+import { StyledBurger } from "./styles";
+import { SidebarContext } from "../../../store/sidebar-context";
 
 const Burger = () => {
   const { open, handleTriggerSidebar } = useContext(SidebarContext);
+  const [scrollPosition, setScrollPosition] = useState(0);
```
Comment (StareMisto) [inline src/components/Sidebar/Burger/index.tsx]: please, use sidebarContext

File: src/components/Sidebar/Burger/index.tsx (hunk 1 above)
Comment (Roksolyanka) [reply src/components/Sidebar/Burger/index.tsx]: corrected

File: src/components/Sidebar/Burger/index.tsx (hunk 2 above)
Comment (Roksolyanka) [reply src/components/Sidebar/Burger/index.tsx]: corrected

**Review by StareMisto — APPROVED** [review]

**Pull request merged**